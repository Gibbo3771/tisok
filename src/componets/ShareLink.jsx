import React from "react";
import { CopyToClipboard } from "react-copy-to-clipboard";

const ShareLink = props => {
  return (
    <CopyToClipboard
      text={`${window.location.href}`}
      onCopy={() => console.log("copied!")}
    >
      <span className="fas fa-clipboard" />
    </CopyToClipboard>
  );
};

export { ShareLink };
