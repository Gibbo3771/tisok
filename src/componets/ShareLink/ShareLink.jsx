// @flow
import React from "react";
import { CopyToClipboard } from "react-copy-to-clipboard";
import ReactTooltip from "react-tooltip";
import { Slip } from "../PropTypes";

export type Props = {
  slip: Slip
};

export default class ShareLink extends React.Component<Props> {
  render() {
    return (
      <React.Fragment>
        <CopyToClipboard text={`${window.location.href}`}>
          <i
            className="fas fa-clipboard share-link"
            ref="clipboard"
            data-for="clipboard"
            data-tip="Copy link to clipboard"
            data-is-capture="true"
          />
        </CopyToClipboard>
        <ReactTooltip
          className="tooltip"
          id="clipboard"
          place="top"
          effect="solid"
          type="tooltip"
          clickable={true}
        />
      </React.Fragment>
    );
  }
}
