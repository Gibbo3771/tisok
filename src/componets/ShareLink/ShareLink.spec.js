import React from "react";
import Adapter from "enzyme-adapter-react-16";
import Enzyme from "enzyme";
import { shallow } from "enzyme";
import ShareLink from "./ShareLink";
import sinon from "sinon";
import { expect as expectChai } from "chai";

Enzyme.configure({ adapter: new Adapter() });

const props = {
  slip: {
    slip_id: null,
    advice: null
  }
};

describe("ShareLink", () => {
  it("renders correctly if it has a slip", () => {
    const { slip } = props;
    const tree = shallow(<ShareLink slip={slip} />);
    expect(tree).toMatchSnapshot();
  });
  it("renders correctly if it has no slip", () => {
    const tree = shallow(<ShareLink />);
    expect(tree).toMatchSnapshot();
  });
});
