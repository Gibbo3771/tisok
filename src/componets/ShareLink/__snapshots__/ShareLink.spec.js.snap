// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShareLink renders correctly if it has a slip 1`] = `ShallowWrapper {}`;

exports[`ShareLink renders correctly if it has no slip 1`] = `ShallowWrapper {}`;
